.dev/scripts/generate_changelog.py

```diff
@@ -0,0 +1,235 @@
+ BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ POSSIBILITY OF SUCH DAMAGE.
+
+"""
+
+import sys
+import json
+import argparse
+from pathlib import Path
+
+import requests
+
+
+CATEGORIES = {
+    "new feature": ("New features", ""),
+    "deprecation": (
+        "Deprecation",
+        "of public APIs, scheduled to be removed after two minor releases",
+    ),
+    "removal": ("Removal", "of the public APIs deprecated in previous releases"),
+    "behavior change": ("Behavior changes", "in classes, apps, or tools",),
+    "API break": (
+        "API changes",
+        "that did not go through the proper deprecation and removal cycle",
+    ),
+    "ABI break": ("ABI changes", "that are still API compatible",),
+    "fix": (None, None),
+    "enhancement": (None, None),
+}
+
+
+MODULES = {
+    "cmake": "CMake",
+    "2d": "libpcl_2d",
+    "common": "libpcl_common",
+    "cuda": "libpcl_cuda",
+    "features": "libpcl_features",
+    "filters": "libpcl_filters",
+    "geometry": "libpcl_geometry",
+    "gpu": "libpcl_gpu",
+    "io": "libpcl_io",
+    "kdtree": "libpcl_kdtree",
+    "keypoints": "libpcl_keypoints",
+    "ml": "libpcl_ml",
+    "octree": "libpcl_octree",
+    "outofcore": "libpcl_outofcore",
+    "people": "libpcl_people",
+    "recognition": "libpcl_recognition",
+    "registration": "libpcl_registration",
+    "sample_consensus": "libpcl_sample_consensus",
+    "search": "libpcl_search",
+    "segmentation": "libpcl_segmentation",
+    "simulation": "libpcl_simulation",
+    "stereo": "libpcl_stereo",
+    "surface": "libpcl_surface",
+    "visualization": "libpcl_visualization",
+    "apps": "PCL Apps",
+    "docs": "PCL Docs",
+    "tutorials": "PCL Tutorials",
+    "examples": "PCL Examples",
+    "tests": "PCL Tests",
+    "tools": "PCL Tools",
+    "ci": "CI",
+    None: "Uncategorized",
+}
+
+
+def fetch_latest_release_date():
+    """
+    Fetch the date when the latest release was created.
+    """
+    url = "https://api.github.com/repos/PointCloudLibrary/pcl/releases/latest"
+    response = requests.get(url)
+    data = response.json()
+    if response.status_code != 200:
+        sys.exit(
+            "Failed to fetch release info. Server response: '{}'".format(
+                data["message"]
+            )
+        )
+    return data["created_at"]
+
+
+def fetch_pr_since(start_date):
+    """
+    Fetch data of PRs merged after a given start date.
+    """
+    url = f"https://api.github.com/search/issues?q=is:pr+repo:PointCloudLibrary/pcl+merged:>={start_date}"
+    pr_data = list()
+    page = 1
+    total_count = 1  # for the first iteration assume that total is 1
+    while len(pr_data) < total_count:
```

Comment:
It's Pythonic to use `while True:... if len(pr_data) == total_count: break` to emulate a `do...while` loop. That eliminates the need for a default value of `total_count`